pysatSpaceWeather/instruments/ace_epam.py

```diff
@@ -0,0 +1,257 @@
+
+Properties
+----------
+platform
+    'ace' Advanced Composition Explorer
+name
+    'epam' Electron, Proton, and Alpha Monitor
+tag
+    - 'realtime' Real-time data from the Space Weather Prediction Center (SWPC)
+    - 'historic' Historic data from the SWPC
+inst_id
+    - ''
+
+Note
+----
+This is not the ACE scientific data set, which will be available at pysatNASA
+
+The real-time data is stored by generation date, where each file contains the
+data for the current day.  If you leave download dates empty, though, it will
+grab today's file three times and assign dates from yesterday, today, and
+tomorrow.
+::
+
+    epam = pysat.Instrument('ace', 'epam', tag='realtime')
+    epam.download(start=epam.today())
+    epam.load(date=epam.today())
+
+
+
+Warnings
+--------
+The real-time data should not be used with the data padding option available
+from pysat.Instrument objects.
+
+"""
+
+import datetime as dt
+import functools
+import numpy as np
+
+from pysat import logger
+
+from pysatSpaceWeather.instruments.methods import ace as mm_ace
+
+# ----------------------------------------------------------------------------
+# Instrument attributes
+
+platform = 'ace'
+name = 'epam'
+tags = {'realtime': 'Real-time data from the SWPC',
+        'historic': ' Historic data from the SWPC'}
+inst_ids = {inst_id: [tag for tag in tags.keys()] for inst_id in ['']}
+
+# Define today's date
+now = dt.datetime.utcnow()
+
+# ----------------------------------------------------------------------------
+# Instrument test attributes
+
+# Set test dates (first level: inst_id, second level: tag)
+_test_dates = {inst_id: {'realtime': dt.datetime(now.year, now.month, now.day),
+                         'historic': dt.datetime(2009, 1, 1)}
+               for inst_id in inst_ids.keys()}
+
+# ----------------------------------------------------------------------------
+# Instrument methods
+
+preprocess = mm_ace.preprocess
+
+
+def init(self):
+    """Initializes the Instrument object with instrument specific values.
+
+    Runs once upon instantiation.
+
+    """
+
+    # Set the appropraite acknowledgements and references
+    self.acknowledgements = mm_ace.acknowledgements()
+    self.references = mm_ace.references(self.name)
+
+    logger.info(self.acknowledgements)
+
+    return
+
+
+def clean(self):
+    """Routine to clean real-time ACE data using the status flag
+
+    Note
+    ----
+    Supports 'clean' and 'dirty'.  Replaces all fill values with NaN.
+    Clean - status flag of zero (nominal data)
+    Dirty - status flag < 9 (accepts bad data record, removes no data record)
+
+    """
+    # Perform the standard ACE cleaning
+    max_status = mm_ace.clean(self)
+
+    # Replace bad values with NaN and remove times with no valid data
+    ecols = ['eflux_38-53', 'eflux_175-315']
+
+    # Evaluate the electron flux data
+    for col in ecols:
+        self.data[col][self.data['status_e'] > max_status] = np.nan
```

Comment:
```suggestion
    self[self.data['status_e'] > max_status, ecols] = np.nan
```
Avoids chained assignment plus the loop.